### PR TITLE
1900 - delete a trial session

### PR DESCRIPTION
**Pull request by codyseibert**

### CODE REVIEW

File: web-api/src/trialSessions/deleteTrialSessionLambda.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const createApplicationContext = require('../applicationContext');
+const {
+  getUserFromAuthHeader,
+  handle,
+} = require('../middleware/apiGatewayHelper');
+
+/**
+ * creates a new trial session.
```
Comment (rachaelparris) [inline web-api/src/trialSessions/deleteTrialSessionLambda.js]: Update jsdoc

Comment (rachaelparris) [inline web-api/src/trialSessions/deleteTrialSessionLambda.js]: Update jsdoc

File: shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    .getPersistenceGateway()
+    .getTrialSessionById({
+      applicationContext,
+      trialSessionId,
+    });
+
+  if (!trialSession) {
+    throw new Error('trial session not found');
+  }
+
```
Comment (rachaelparris) [inline shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.js:37]: We shouldn't delete a trial session that has already passed, right? Should you also check for that here and throw an error?

File: shared/src/persistence/dynamo/trialSessions/deleteTrialSessionWorkingCopy.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const client = require('../../dynamodbClientService');
+
+/**
+ * deleteTrialSessionWorkingCopy
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @param {object} providers.trialSessionWorkingCopy the trial session working copy data
```
Comment (kkoskelin) [inline shared/src/persistence/dynamo/trialSessions/deleteTrialSessionWorkingCopy.js]: *jsdoc*

**Review by kkoskelin — CHANGES_REQUESTED** [review]
Once jsdoc is updated and we're checking for trial sessions' date against current date, good to go.

**Review by kkoskelin — APPROVED** [review]

File: shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.test.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        getTrialSessionById: () => null,
+      }),
+    };
+
+    await expect(
+      deleteTrialSessionInteractor({
+        applicationContext,
+        trialSessionId: 'c54ba5a9-b37b-479d-9201-067ec6e335bb',
+      }),
+    ).rejects.toThrow('trial session not found');
```
Comment (matthopson) [inline shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.test.js:63]: I feel like we don't use this pattern enough. 👏

Comment (matthopson) [reply shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.test.js:63]: (it's usually a try/catch, and then setting an error)

File: shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.test.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        getCaseByCaseId: () => MOCK_CASE,
+        getTrialSessionById: () => ({
+          ...MOCK_TRIAL,
+          startDate: '2100-12-01T00:00:00.000Z',
+        }),
+        updateCase: updateCaseSpy,
+      }),
+      getUniqueId: () => 'c54ba5a9-b37b-479d-9201-067ec6e335bb',
+      getUtilities: () => ({
+        createISODateString: () => '2050-12-01T00:00:00.000Z',
```
Comment (matthopson) [inline shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.test.js:120]: Very brittle. This code is going to fail in 40 years.

**Review by rachaelparris — APPROVED** [review]

**Review by matthopson — APPROVED** [review]

**Pull request merged**